### PR TITLE
cdo 2.4.0

### PR DESCRIPTION
**Pull request by p-linnane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/cdo.rb [hunk 1]
```diff
@@ -27,17 +27,24 @@ class Cdo < Formula
   depends_on "proj"
   uses_from_macos "python" => :build
 
+  on_macos do
+    depends_on "llvm" => :build if DevelopmentTools.clang_build_version <= 1403
+  end
```
Comment (ZhongRuoyu) [inline Formula/c/cdo.rb:32]: To actually use this we also need

```ruby
ENV.llvm_clang if OS.mac? && DevelopmentTools.clang_build_version <= 1400
```

in the `install` method.

Comment (p-linnane) [reply Formula/c/cdo.rb:32]: Thanks. Not sure how I forgot that 🤦‍♂️

File: Formula/c/cdo.rb [hunk 2]
```diff
@@ -27,17 +27,26 @@ class Cdo < Formula
...
+
+  fails_with :clang do
+    build 1403
+    cause "Requires C++20 support"
+  end
+
   def install
+    ENV.llvm_clang if OS.mac? && DevelopmentTools.clang_build_version <= 1403
+
     args = %W[
```
Comment (ZhongRuoyu) [inline Formula/c/cdo.rb:42]: It automatically linked to OpenMP due to LLVM. We can disable that with:

```suggestion
    args = %W[
      --disable-openmp
```

**Review by ZhongRuoyu — APPROVED** [review]

**Comment by p-linnane** [comment]
Thanks @ZhongRuoyu! 🙏

**Pull request merged**